front/components/agent_builder/AgentBuilderFormContext.tsx

```diff
@@ -176,8 +208,20 @@ export const agentBuilderFormSchema = z.object({
   actions: z.array(actionSchema),
 });
 
+export const agentBuilderMCPFormSchema = z.object({
+  agentSettings: agentSettingsSchema,
+  instructions: z.string().min(1, "Instructions are required"),
+  generationSettings: generationSettingsSchema,
+  actions: z.array(mcpActionSchema),
+});
+
 export type AgentBuilderFormData = z.infer<typeof agentBuilderFormSchema>;
 
+// TODO: have only one AgentBuilderAction.
 export type AgentBuilderAction = z.infer<typeof actionSchema>;
+export type AgentBuilderMCPAction = z.infer<typeof mcpActionSchema>;
+export type AgentBuilderDataVizAction = z.infer<
+  typeof dataVisualizationActionSchema
+>;
```

Comment:
Noob question: what's the limitation here to only use one `AgentBuilderAction` as we were doing before ?